lib/zmq-kernel.js

```diff
@@ -429,7 +429,7 @@ export default class ZMQKernel extends Kernel {
 
     if (this.kernelProcess) {
       this._kill();
-      fs.unlink(this.connectionFile);
+      fs.unlinkSync(this.connectionFile);
```

Comment:
I think we unlinkSync is better, it's not so slow just removing one file.
When destroy() finished, I want assume all cleanup stuff done. 
IMO reliable/simple behavior is important for shutdown.
Then if performance is really matter, we then move to tune that part like destroy return promise.